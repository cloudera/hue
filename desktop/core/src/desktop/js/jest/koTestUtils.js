// Licensed to Cloudera, Inc. under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  Cloudera, Inc. licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as ko from 'knockout';

export const koSetup = () => {
  let originalLoadTemplate;
  let wrapper;

  beforeEach(() => {
    originalLoadTemplate = ko.components.defaultLoader.loadTemplate;
    ko.components.defaultLoader.loadTemplate = (name, templateConfig, callback) => {
      const div = window.document.createElement('div');
      div.innerHTML = templateConfig;
      callback(div.children);
    };

    wrapper = window.document.createElement('div');
    wrapper.classList.add('component-wrapper');
    window.document.querySelector('body').appendChild(wrapper);
  });

  afterEach(() => {
    ko.components.defaultLoader.loadTemplate = originalLoadTemplate;
    wrapper.parentNode.removeChild(wrapper);
    ko.cleanNode(wrapper);
    expect(window.document.querySelectorAll('[data-test]').length).toEqual(0);
  });

  return {
    renderComponent: async (name, data, instantTimeout) =>
      new Promise((resolve, reject) => {
        const element = window.document.createElement('div');
        element.setAttribute(
          'data-bind',
          `descendantsComplete: descendantsComplete, component: { name: "${name}", params: params }`
        );
        wrapper.appendChild(element);

        try {
          ko.components.get(name, comp => {
            if (comp.createViewModel) {
              const origSetTimeout = window.setTimeout;
              if (instantTimeout) {
                window.setTimeout = cb => cb();
                comp.createViewModel(data, { element: element });
                window.setTimeout = origSetTimeout;
              } else {
                comp.createViewModel(data, { element: element });
              }

              ko.applyBindings(
                {
                  descendantsComplete: () => resolve(wrapper),
                  params: data
                },
                wrapper
              );
            } else {
              reject('no createViewModel function found on component ' + name);
            }
          });
        } catch (e) {
          reject(e);
        }
      }),

    renderKo: async (html, viewModel) =>
      new Promise(resolve => {
        const setupContainer = window.document.createElement('div');
        setupContainer.setAttribute(
          'data-bind',
          '{ descendantsComplete: descendantsComplete, with: model }'
        );

        const container = window.document.createElement('div');
        container.innerHTML = html;
        setupContainer.appendChild(container);

        const data = {
          model: viewModel,
          descendantsComplete: () => resolve(container)
        };

        ko.applyBindings(data, setupContainer);
      }),

    waitForKoUpdate: async () =>
      new Promise(resolve => {
        window.setTimeout(resolve, 0);
      })
  };
};

export const waitForObservableChange = async observable =>
  new Promise(resolve => {
    const sub = observable.subscribe(() => {
      sub.dispose();
      resolve();
    });
  });

export const simulateInput = (element, text) => {
  element.value = text;
  element.dispatchEvent(new Event('change'));
};
