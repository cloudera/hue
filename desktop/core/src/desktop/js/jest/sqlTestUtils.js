// Licensed to Cloudera, Inc. under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  Cloudera, Inc. licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Needed to compare by val without taking attr order into account
const resultEquals = function(a, b) {
  if (typeof a !== typeof b) {
    return false;
  }
  if (a === b) {
    return true;
  }
  if (typeof a === 'object') {
    const aKeys = Object.keys(a);
    if (aKeys.length !== Object.keys(b).length) {
      return false;
    }
    for (let i = 0; i < aKeys.length; i++) {
      if (!resultEquals(a[aKeys[i]], b[aKeys[i]])) {
        return false;
      }
    }
    return true;
  }
  return a == b;
};

const jsonStringToJsString = function(jsonString) {
  return jsonString
    .replace(/'([a-zA-Z]+)':/g, (all, group) => {
      return group + ':';
    })
    .replace(/([:{,])/g, (all, group) => {
      return group + ' ';
    })
    .replace(/[}]/g, ' }')
    .replace(/["]/g, "'")
    .replace(/'([a-z_]+)':/gi, '$1:');
};

expect.extend({
  toEqualAutocompleteValues(actualItems, expectedValues) {
    if (actualItems.length !== expectedValues.length) {
      return { pass: false };
    }

    for (let i = 0; i < expectedValues.length; i++) {
      const stringValue =
        typeof actualItems[i] !== 'string' ? '' + actualItems[i].value : actualItems[i].value;
      if (stringValue !== expectedValues[i]) {
        return { pass: false };
      }
    }
    return { pass: true };
  },
  toEqualDefinition(actualResponse, testDefinition) {
    if (typeof testDefinition.noErrors === 'undefined' && actualResponse.errors) {
      let allRecoverable = true;
      actualResponse.errors.forEach(error => {
        allRecoverable = allRecoverable && error.recoverable;
      });
      if (allRecoverable) {
        delete actualResponse.errors;
      }
    }

    if (
      ((testDefinition.expectedResult && testDefinition.expectedResult.locations) ||
        testDefinition.expectedLocations) &&
      actualResponse.locations
    ) {
      const expectedLoc =
        testDefinition.expectedLocations || testDefinition.expectedResult.locations;
      const expectsType = expectedLoc.some(location => location.type === 'statementType');
      if (!expectsType) {
        actualResponse.locations = actualResponse.locations.filter(
          location => location.type !== 'statementType'
        );
      }
    }

    if (testDefinition.expectedDefinitions) {
      if (!resultEquals(actualResponse.definitions, testDefinition.expectedDefinitions)) {
        return {
          pass: false,
          message: () =>
            '\n        Statement: ' +
            testDefinition.beforeCursor +
            '|' +
            testDefinition.afterCursor +
            '\n' +
            'Expected definitions: ' +
            jsonStringToJsString(JSON.stringify(testDefinition.expectedDefinitions)) +
            '\n' +
            '  Parser definitions: ' +
            jsonStringToJsString(JSON.stringify(actualResponse.definitions)) +
            '\n'
        };
      }
    } else {
      delete actualResponse.definitions;
    }

    if (testDefinition.locationsOnly) {
      return {
        pass: resultEquals(actualResponse.locations, testDefinition.expectedLocations),
        message: () =>
          '\n        Statement: ' +
          testDefinition.beforeCursor +
          '|' +
          testDefinition.afterCursor +
          '\n' +
          'Expected locations: ' +
          jsonStringToJsString(JSON.stringify(testDefinition.expectedLocations)) +
          '\n' +
          '  Parser locations: ' +
          jsonStringToJsString(JSON.stringify(actualResponse.locations)) +
          '\n'
      };
    }

    if (actualResponse.suggestKeywords) {
      const weightFreeKeywords = [];
      actualResponse.suggestKeywords.forEach(keyword => {
        weightFreeKeywords.push(keyword.value);
      });
      actualResponse.suggestKeywords = weightFreeKeywords;
    }

    if (!!testDefinition.noLocations) {
      if (actualResponse.locations.length > 0) {
        return {
          pass: false,
          message: () =>
            '\nStatement: ' +
            testDefinition.beforeCursor +
            '|' +
            testDefinition.afterCursor +
            '\n' +
            '           Expected no locations, found ' +
            actualResponse.locations.length
        };
      }
    }
    if (typeof testDefinition.expectedResult.locations === 'undefined') {
      delete actualResponse.locations;
    }
    let deleteKeywords = false;
    if (testDefinition.containsColRefKeywords) {
      if (typeof actualResponse.suggestColRefKeywords == 'undefined') {
        return {
          pass: false,
          message: () =>
            '\nStatement: ' +
            testDefinition.beforeCursor +
            '|' +
            testDefinition.afterCursor +
            '\n' +
            '           No colRef keywords found'
        };
      } else if (testDefinition.containsColRefKeywords !== true) {
        let contains = true;
        testDefinition.containsColRefKeywords.forEach(keyword => {
          contains =
            contains &&
            (actualResponse.suggestColRefKeywords.BOOLEAN.indexOf(keyword) !== -1 ||
              actualResponse.suggestColRefKeywords.NUMBER.indexOf(keyword) !== -1 ||
              actualResponse.suggestColRefKeywords.STRING.indexOf(keyword) !== -1);
        });
        if (!contains) {
          return {
            pass: false,
            message: () =>
              '\nStatement: ' +
              testDefinition.beforeCursor +
              '|' +
              testDefinition.afterCursor +
              '\n' +
              '           Expected colRef keywords not found ' +
              'Expected keywords: ' +
              JSON.stringify(testDefinition.containsColRefKeywords) +
              '\n' +
              '  Parser keywords: ' +
              JSON.stringify(actualResponse.suggestColRefKeywords) +
              '\n'
          };
        }
      }
      delete actualResponse.suggestColRefKeywords;
    }

    if (typeof testDefinition.containsKeywords !== 'undefined') {
      const keywords = actualResponse.suggestKeywords;
      let contains = true;
      testDefinition.containsKeywords.forEach(keyword => {
        if (typeof keywords === 'undefined' || keywords.indexOf(keyword) === -1) {
          contains = false;
          return false;
        }
      });
      if (!contains) {
        return {
          pass: false,
          message: () =>
            '\n        Statement: ' +
            testDefinition.beforeCursor +
            '|' +
            testDefinition.afterCursor +
            '\n' +
            'Expected keywords: ' +
            JSON.stringify(testDefinition.containsKeywords) +
            '\n' +
            '  Parser keywords: ' +
            JSON.stringify(keywords) +
            '\n'
        };
      }
      deleteKeywords = true;
    }
    if (typeof testDefinition.doesNotContainKeywords !== 'undefined') {
      const keywords = actualResponse.suggestKeywords || [];
      let contains = false;
      testDefinition.doesNotContainKeywords.forEach(keyword => {
        if (typeof keywords === 'undefined' || keywords.indexOf(keyword) !== -1) {
          contains = true;
          return false;
        }
      });
      if (contains) {
        return {
          pass: false,
          message: () =>
            '\n            Statement: ' +
            testDefinition.beforeCursor +
            '|' +
            testDefinition.afterCursor +
            '\n' +
            'Not expected keywords: ' +
            JSON.stringify(testDefinition.doesNotContainKeywords) +
            '\n' +
            '      Parser keywords: ' +
            JSON.stringify(keywords) +
            '\n'
        };
      }
      deleteKeywords = true;
    }

    if (deleteKeywords) {
      delete actualResponse.suggestKeywords;
    }
    return {
      pass: resultEquals(actualResponse, testDefinition.expectedResult),
      message: () =>
        '\n        Statement: ' +
        testDefinition.beforeCursor +
        '|' +
        testDefinition.afterCursor +
        '\n' +
        'Expected response: ' +
        jsonStringToJsString(JSON.stringify(testDefinition.expectedResult) + '\n') +
        '  Parser response: ' +
        jsonStringToJsString(JSON.stringify(actualResponse) + '\n')
    };
  }
});
