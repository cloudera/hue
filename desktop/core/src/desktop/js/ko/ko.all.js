// Licensed to Cloudera, Inc. under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  Cloudera, Inc. licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as ko from 'knockout';
import 'ko/ko.init';
import komapping from 'knockout.mapping';
import 'knockout-switch-case';
import 'knockout-sortable';
import 'knockout.validation';

import 'ext/ko.editable.custom';
import 'ext/ko.selectize.custom';

import 'ko/bindings/charts/mapchart/ko.mapChart';
import 'ko/bindings/charts/ko.barChart';
import 'ko/bindings/charts/ko.leafletMapChart';
import 'ko/bindings/charts/ko.lineChart';
import 'ko/bindings/charts/ko.partitionChart';
import 'ko/bindings/charts/ko.pieChart';
import 'ko/bindings/charts/ko.scatterChart';
import 'ko/bindings/charts/ko.timelineChart';

import 'ko/bindings/ko.vue.js';

import 'ko/bindings/ace/ko.aceEditor';
import 'ko/bindings/ace/ko.aceResizer';
import 'ko/bindings/ko.appAwareTemplateContextMenu';
import 'ko/bindings/ko.assistFileDraggable';
import 'ko/bindings/ko.assistFileDroppable';
import 'ko/bindings/ko.assistVerticalResizer';
import 'ko/bindings/ko.attachViewModelToElementData';
import 'ko/bindings/ko.augmentHtml';
import 'ko/bindings/ko.autocomplete';
import 'ko/bindings/ko.autogrowInput';
import 'ko/bindings/ko.blurHide';
import 'ko/bindings/ko.bytesize';
import 'ko/bindings/ko.chosen';
import 'ko/bindings/ko.clearable';
import 'ko/bindings/ko.clickForAceFocus';
import 'ko/bindings/ko.clickToCopy';
import 'ko/bindings/ko.clipboard';
import 'ko/bindings/ko.codemirror';
import 'ko/bindings/ko.contextMenu';
import 'ko/bindings/ko.contextSubMenu';
import 'ko/bindings/ko.datalist';
import 'ko/bindings/ko.datepicker';
import 'ko/bindings/ko.dateRangePicker';
import 'ko/bindings/ko.dblClick';
import 'ko/bindings/ko.delayedOverflow';
import 'ko/bindings/ko.dockable';
import 'ko/bindings/ko.documentChooser';
import 'ko/bindings/ko.documentContextPopover';
import 'ko/bindings/ko.draggableText';
import 'ko/bindings/ko.dropdown';
import 'ko/bindings/ko.dropzone';
import 'ko/bindings/ko.duration';
import 'ko/bindings/ko.ellipsis';
import 'ko/bindings/ko.fadeVisible';
import 'ko/bindings/ko.fetchMore';
import 'ko/bindings/ko.fileChooser';
import 'ko/bindings/ko.foreachVisible';
import 'ko/bindings/ko.fresherEditor';
import 'ko/bindings/ko.hdfsAutocomplete';
import 'ko/bindings/ko.hdfsTree';
import 'ko/bindings/ko.highlight';
import 'ko/bindings/ko.hiveChooser';
import 'ko/bindings/ko.html';
import 'ko/bindings/ko.hueach';
import 'ko/bindings/ko.hueCheckAll';
import 'ko/bindings/ko.hueCheckbox';
import 'ko/bindings/ko.hueChecked';
import 'ko/bindings/ko.hueLink';
import 'ko/bindings/ko.hueSpinner';
import 'ko/bindings/ko.impalaDagre';
import 'ko/bindings/ko.jHueRowSelector';
import 'ko/bindings/ko.logResizer';
import 'ko/bindings/ko.logScroller';
import 'ko/bindings/ko.medium';
import 'ko/bindings/ko.moment';
import 'ko/bindings/ko.momentFromNow';
import 'ko/bindings/ko.multiCheck';
import 'ko/bindings/ko.multiCheckForeachVisible';
import 'ko/bindings/ko.multiClick';
import 'ko/bindings/ko.multiLineEllipsis';
import 'ko/bindings/ko.numberFormat';
import 'ko/bindings/ko.numericTextInput';
import 'ko/bindings/ko.onClickOutside';
import 'ko/bindings/ko.oneClickSelect';
import 'ko/bindings/ko.parseArguments';
import 'ko/bindings/ko.publish';
import 'ko/bindings/ko.reactWrapper';
import 'ko/bindings/ko.readOnlyAce';
import 'ko/bindings/ko.resizable';
import 'ko/bindings/ko.select2';
import 'ko/bindings/ko.simplesize';
import 'ko/bindings/ko.slider';
import 'ko/bindings/ko.slideVisible';
import 'ko/bindings/ko.solrChooser';
import 'ko/bindings/ko.spinEdit';
import 'ko/bindings/ko.splitDraggable';
import 'ko/bindings/ko.splitFlexDraggable';
import 'ko/bindings/ko.sqlContextPopover';
import 'ko/bindings/ko.stickVertical';
import 'ko/bindings/ko.storageContextPopover';
import 'ko/bindings/ko.stretchDown';
import 'ko/bindings/ko.tagEditor';
import 'ko/bindings/ko.tagsNotAllowed';
import 'ko/bindings/ko.templateContextMenu';
import 'ko/bindings/ko.templatePopover';
import 'ko/bindings/ko.textSqueezer';
import 'ko/bindings/ko.toggle';
import 'ko/bindings/ko.toggleOverflow';
import 'ko/bindings/ko.tooltip';
import 'ko/bindings/ko.truncatedText';
import 'ko/bindings/ko.typeahead';
import 'ko/bindings/ko.verticalSlide';
import 'ko/bindings/ko.visibleOnHover';

import 'ko/components/appSwitcher/ko.appSwitcher';
import 'ko/components/assist/ko.assistDashboardPanel';
import 'ko/components/assist/ko.assistDbPanel';
import 'ko/components/assist/ko.assistDocumentsPanel';
import 'ko/components/assist/ko.assistEditorContextPanel';
import 'ko/components/assist/ko.assistFunctionsPanel';
import 'ko/components/assist/ko.assistGitPanel';
import 'ko/components/assist/ko.assistHBasePanel';
import 'ko/components/assist/ko.assistLangRefPanel';
import 'ko/components/assist/ko.assistPanel';
import 'ko/components/assist/ko.assistSchedulePanel';
import 'ko/components/assist/ko.assistStoragePanel';
import 'ko/components/assist/ko.rightAssistPanel';
import 'ko/components/contextPopover/ko.contextPopover';
import 'ko/components/contextPopover/ko.quickQueryContext';
import 'ko/components/contextPopover/ko.popoverOnHover';
import 'ko/components/doc/ko.shareDocModal';
import 'ko/components/simpleAceEditor/ko.simpleAceEditor';

import 'ko/components/ko.catalogEntriesList';
import 'ko/components/ko.contextSelector';
import 'ko/components/ko.createDirectoryModal';
import 'ko/components/ko.deleteDocModal';
import 'ko/components/ko.dropDown';
import 'ko/components/ko.editorDroppableMenu';
import 'ko/components/ko.executionAnalysis';
import 'ko/components/ko.favoriteApp';
import 'ko/components/ko.fieldSamples';
import 'ko/components/ko.globalSearch';
import 'ko/components/ko.historyPanel';
import 'ko/components/ko.importDocumentsModal';
import 'ko/components/ko.inlineAutocomplete';
import 'ko/components/ko.jobBrowserLinks';
import 'ko/components/ko.navProperties';
import 'ko/components/ko.navTags';
import 'ko/components/ko.pollingCatalogEntriesList';
import 'ko/components/ko.quickQueryAction';
import 'ko/components/ko.sentryPrivileges';
import 'ko/components/ko.sessionAuthModal';
import 'ko/components/ko.sessionPanel';
import 'ko/components/ko.shareGistModal';
import 'ko/components/ko.sqlColumnsTable';

// TODO: Move to about app when it has it's own webpack entry
import 'apps/about/components/ko.hueConfigTree';
import 'apps/about/components/ko.connectorsConfig';

import 'ko/extenders/ko.maxLength';
import 'ko/extenders/ko.numeric';
import 'ko/extenders/ko.toJson';

import 'ko/observables/ko.observableArrayDefault';
import 'ko/observables/ko.observableDefault';

window.ko = ko;
window.ko.mapping = komapping;

export default ko;
