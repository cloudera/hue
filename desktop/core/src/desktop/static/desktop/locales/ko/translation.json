{
  "Add": "추가",
  "aggregate": "집계",
  "Aggregate": "집계",
  "alias": "별칭",
  "All": "모두",
  "and": "및",
  "Apps": "앱",
  "Assist": "지원",
  "Assistant": "도우미",
  "Back": "뒤로",
  "Browsers": "브라우저",
  "Bundle": "Bundle",
  "Cancel": "취소",
  "Change": "변경",
  "Clipboard": "클립보드",
  "Close": "닫기",
  "Clusters": "클러스터",
  "Columns": "열",
  "condition": "조건",
  "Create": "생성",
  "Created": "생성됨",
  "CSV": "CSV",
  "cte": "cte",
  "CTEs": "CTE",
  "Dashboard": "대시보드",
  "database": "데이터베이스",
  "Databases": "데이터베이스",
  "Date": "날짜",
  "default": "기본값:",
  "Delete": "삭제",
  "Description": "설명",
  "Details": "세부 정보",
  "Directory": "디렉토리",
  "Documentation": "문서",
  "Documents": "문서",
  "Editor": "편집기",
  "Excel": "Excel",
  "Export": "내보내기",
  "Failed": "실패함",
  "Field": "필드",
  "File": "파일",
  "Files": "파일",
  "filter": "필터",
  "Filter": "필터",
  "Format": "형식",
  "Functions": "함수",
  "Group": "그룹",
  "HBase": "HBase",
  "Help": "도움말",
  "Hive": "Hive",
  "Identifiers": "식별자",
  "Impala": "Impala",
  "Import": "가져오기",
  "Indexes": "인덱스",
  "Insert": "커서 위치에 ",
  "Jobs": "Job",
  "join": "가입",
  "Key": "키",
  "keyword": "키워드",
  "Keywords": "키워드",
  "Metrics": "메트릭",
  "Modify": "수정",
  "More": "더",
  "Name": "이름",
  "Next": "다음",
  "No": "아니요",
  "Notebook": "노트북",
  "of": "/",
  "Open": "열기",
  "Options": "옵션",
  "Order": "순서",
  "other": "기타",
  "Output": "출력",
  "Owner": "소유자",
  "Partitions": "파티션",
  "Password": "암호",
  "Permissions": "권한",
  "Popular": "인기",
  "Popularity": "인기도",
  "Project": "프로젝트",
  "Queries": "쿼리",
  "Read": "읽기",
  "Refresh": "새로 고침",
  "Remove": "제거",
  "Report": "보고서",
  "Running": "실행 중",
  "Sample": "샘플",
  "sample": "샘플",
  "Samples": "샘플",
  "Schedule": "예약",
  "Search": "검색",    
  "Settings": "설정",
  "Share": "공유",
  "Sharing": "공유",
  "Size": "크기",
  "Sources": "소스",
  "SQL": "SQL",
  "Start": "시작",
  "Statement": "명령문",
  "Stats": "상태",
  "Stopped": "중지됨",
  "table": "테이블",
  "Table": "테이블",
  "Tables": "테이블",
  "Terms": "용어",
  "Type": "유형",
  "UDFs": "UDF",
  "Undo": "실행 취소",
  "Username": "사용자 이름",
  "Value": "값",
  "Values": "값",
  "variable": "변수",
  "Variables": "변수",
  "View": "보기",
  "view": "보기",
  "Views": "보기",
  "virtual": "가상",
  "Workflow": "Workflow",
  "Yes": "예"
}