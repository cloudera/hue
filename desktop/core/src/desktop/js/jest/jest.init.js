// Licensed to Cloudera, Inc. under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  Cloudera, Inc. licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Ensure singletons
import 'core-js/stable';
import 'regenerator-runtime/runtime';

import 'apps/editor/execution/sessionManager';
import './jquery.setup';
import './sqlTestUtils';

import 'ext/bootstrap.2.3.2.min';
import axios from 'axios';
import $ from 'jquery';
import * as ko from 'knockout';
import komapping from 'knockout.mapping';

ko.mapping = komapping;

class Tooltip {}

class AceRange {}

class Autocomplete {}

const globalVars = {
  ko: ko,
  AUTOCOMPLETE_TIMEOUT: 1,
  CACHEABLE_TTL: 1,
  HAS_LINK_SHARING: true,
  HAS_SQL_ANALYZER: false,
  HUE_I18n: {},
  HUE_BASE_URL: '',
  HUE_CHARTS: {
    TYPES: 'bar'
  },
  LOGGED_USERNAME: 'foo',
  LOGGED_USER_ID: 'bar',
  PREVENT_AUTOFILL_INPUT_ATTRS:
    'autocorrect="off" autocomplete="do-not-autocomplete" autocapitalize="off" spellcheck="false"',
  STATIC_URLS: {
    'impala/art/icon_impala_48.png': 'impala/art/icon_impala_48.png',
    'beeswax/art/icon_beeswax_48.png': 'beeswax/art/icon_beeswax_48.png'
  },
  SQL_COLUMNS_KNOWN_FACET_VALUES: {
    type: { array: -1, boolean: -1 }
  },
  ace: {
    edit: () => ({
      setOptions: () => {},
      getSession: () => ({
        setMode: () => {},
        doc: {
          createAnchor: () => ({
            on: () => {},
            getPosition: () => ({
              row: 0
            }),
            setPosition: () => {},
            detach: () => {}
          })
        },
        getTextRange: () => {},
        addGutterDecoration: () => {},
        removeGutterDecoration: () => {}
      }),
      setTheme: () => {},
      getValue: () => '',
      getSelectionRange: () => ({ start: { row: 0, column: 0 }, end: { row: 0, column: 0 } }),
      on: () => {},
      off: () => {},
      commands: {
        on: () => {},
        off: () => {}
      },
      container: {
        addEventListener: () => {},
        removeEventListener: () => {}
      }
    }),
    require: () => ({
      Tooltip: Tooltip,
      Range: AceRange,
      Autocomplete: Autocomplete
    })
  }
};

Object.keys(globalVars).forEach(key => {
  global[key] = globalVars[key];
  global.window[key] = globalVars[key];
});

$.ajaxSetup({
  beforeSend: function () {
    console.warn('actual jQuery ajax called');
    console.trace();
  }
});

axios.interceptors.request.use(config => {
  console.warn('Actual axios ajax request made to url: ' + config.url);
  console.trace();
  return config;
});

process.on('unhandledRejection', err => {
  fail(err);
});

jest.mock('../utils/i18nReact');
jest.mock('../utils/hueAnalytics');
