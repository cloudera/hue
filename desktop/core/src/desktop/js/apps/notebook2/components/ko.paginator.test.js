import { koSetup } from 'jest/koTestUtils';
import { NAME } from './ko.paginator';

describe('ko.paginator.js', () => {
  const setup = koSetup();

  it('should render component', async () => {
    const element = await setup.renderComponent(NAME, {
      totalPages: () => 5,
      currentPage: () => 1
    });

    expect(element.innerHTML).toMatchSnapshot();
  });
});
