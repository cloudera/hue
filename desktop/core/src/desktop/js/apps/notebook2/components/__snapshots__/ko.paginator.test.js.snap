// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ko.paginator.js should render component 1`] = `
"<div data-bind=\\"descendantsComplete: descendantsComplete, component: { name: &quot;paginator&quot;, params: params }\\"><div class=\\"pagination\\" data-bind=\\"visible: totalPages() > 1\\" style=\\"\\">
  <ul>
    <li data-bind=\\"css: { 'disabled' : currentPage() === 1 }\\" class=\\"disabled\\">
      <a href=\\"javascript: void(0);\\" data-bind=\\"click: previousPage.bind($data)\\">Prev</a>
    </li>
    <li class=\\"active\\"><span data-bind=\\"text: currentPage() + '/' + totalPages()\\">1/5</span></li>
    <li data-bind=\\"css: { 'disabled' : currentPage() === totalPages() }\\">
      <a href=\\"javascript: void(0);\\" data-bind=\\"click: nextPage.bind($data)\\">Next</a>
    </li>
  </ul>
</div></div>"
`;
